watchtower/wtclient/client.go

```diff
@@ -1069,6 +1090,53 @@ func (c *client) initActiveQueue(s *ClientSession,
 	return sq
 }
 
+// terminateSession sets the given session's status to CSessionTerminal meaning
+// that it will not be used again.
+func (c *client) terminateSession(id *wtdb.SessionID) error {
+	errChan := make(chan error, 1)
+
+	select {
+	case c.terminateSessions <- &terminateSessMsg{
+		id:      id,
+		errChan: errChan,
+	}:
+	case <-c.pipeline.quit:
+		return ErrClientExiting
+	}
+
+	select {
+	case err := <-errChan:
+		return err
+	case <-c.pipeline.quit:
+		return ErrClientExiting
+	}
+}
+
+// handleTerminateSession handles a request to terminate a session. It will
+// first shut down the session if it is part of the active session set, then
+// it will ensure that the active session queue is set reset if it is using the
+// session in question. Finally, the session's status in the DB will be updated.
+func (c *client) handleTerminateSession(msg *terminateSessMsg) error {
+	id := *msg.id
```

Comment:
Do we need a `nil` check here? Or in `terminateSession`? Or not use a pointer as we do for the tower?

Reply:
fixed!